### PR TITLE
Fixed Rules for Racism

### PR DESCRIPTION
**Pull request by mattHaynie**
Added more Slurs to check for racist users to easily detect them in the future.

### CODE REVIEW

**Comment by SharpOB** [comment]
You've done a fantastic job of adding a bunch of potential false positives. 'Spook', 'Gringo', 'Jew', '13%' and others. 'MattHaynie' is an interesting one.
Go back to Reddit.

**Comment by mattHaynie** [comment]
> You've done a fantastic job of adding a bunch of potential false positives. 'Spook', 'Gringo', 'Jew', '13%' and others. 'MattHaynie' is an interesting one.
> Go back to Reddit.

Im not sure how "Going back to Reddit" is going to help anyone here, but i tried and had a few laughs on r/Funny (/s!). Anyways id like to address these additions, 'Spook' is a common term to refer to a person of color (this is a very old term) and hopefully (i didnt check) the regex being used to identify these words only checks the words and not sub-words. i can understand the confusion here because you might want to avoid flagging someone for saying 'spooky' and i agree. of course with all other changes in this PR these are up to the owner and are merely a suggestion. 'Gringo' is a slur, no debate here. "Jew" is rarely used in actual conversation about people of jewish faith because of the negative connotation surrounding it when used by alt-right white supremacists. the correct term is "person of jewish faith" and if referring as a verb "jewish". "13%" is a common white supremacist dogwhistle used by members of the alt right to refer to a (long outdated and racist) statistic supporting the idea that male persons of color make up 13% of the american population but commit 50% of the crimes(yes they actually believe this). 

As i said before, all these changes are up for vetting by the author of this software, please accept my changes with a grain of salt and feel free to remove what you dont agree with. I wont be offended, others might, but ill be fine. 
Thanks for reading!

**Comment by hemlotheregamerfrens** [comment]
how about we just dont.
this is a terrible idea, wordfilters do not work and 90% of people that play video games shrug off racial slurs to begin with regardless of race. this is a non issue.

**Comment by hemlotheregamerfrens** [comment]
> Looks great!

did u actually look at it

**Comment by kittenzok** [comment]
> > Looks great!
> 
> did u actually look at it

Yes, i have! I think its a fairly complete list of racial slurs, maybe a few could be removed to prevent false positives but overall the idea is great. The original filter would not suffice.
Regarding your previous comment condoning online hate in video games, that is disgusting and you should be ashamed of yourself. please consider taking the anykey pledge: https://www.anykey.org/en/pledge

**Comment by hemlotheregamerfrens** [comment]
> > > Looks great!
> > 
> > 
> > did u actually look at it
> 
> Yes, i have! I think its a fairly complete list of racial slurs, maybe a few could be removed to prevent false positives but overall the idea is great. The original filter would not suffice.
> Regarding your previous comment condoning online hate in video games, that is disgusting and you should be ashamed of yourself. please consider taking the anykey pledge: https://www.anykey.org/en/pledge

ur rite gamer
what was i thinking

**Comment by SharpOB** [comment]
> 
> 
> > You've done a fantastic job of adding a bunch of potential false positives. 'Spook', 'Gringo', 'Jew', '13%' and others. 'MattHaynie' is an interesting one.
> > Go back to Reddit.
> 
> Im not sure how "Going back to Reddit" is going to help anyone here, but i tried and had a few laughs on r/Funny (/s!). Anyways id like to address these additions, 'Spook' is a common term to refer to a person of color (this is a very old term) and hopefully (i didnt check) the regex being used to identify these words only checks the words and not sub-words. i can understand the confusion here because you might want to avoid flagging someone for saying 'spooky' and i agree. of course with all other changes in this PR these are up to the owner and are merely a suggestion. 'Gringo' is a slur, no debate here. "Jew" is rarely used in actual conversation about people of jewish faith because of the negative connotation surrounding it when used by alt-right white supremacists. The use of the term can be quite disgusting, and i have personally been using this list while running TF2BD and have had 0 false flags. Heres one of the messages caught by the system: [Image](https://archive-media-0.nyafuu.org/bant/image/1510/78/1510783955112.jpg), as you can see, its rarely used in actual reference to people of the jewish faith but rather used as an insult. please refer to the ADL's page on this aswell. correct term is "person of jewish faith" and if referring as a verb "jewish". "13%" is a common white supremacist dogwhistle used by members of the alt right to refer to a (long outdated and racist) statistic supporting the idea that male persons of color make up 13% of the american population but commit 50% of the crimes(yes they actually believe this).
> 
> As i said before, all these changes are up for vetting by the author of this software, please accept my changes with a grain of salt and feel free to remove what you dont agree with. I wont be offended, others might, but ill be fine.
> Thanks for reading!

Why did you send an image of a naked black man, and why is 'MattHaynie' a slur? Additionally, the '13%' dogwhistle is completely legitimate, it's real crime statistics, and obviously there is a LOT of potential for all of these to come up in normal conversation, for example, it is always racist to say 'nigger', it depends on the context. Context is completely ignored, so implementing this functionality will always have false positives.

File: tf2_bot_detector/cfg/rules.official.json [hunk 1]
```diff
@@ -108,7 +108,40 @@
...
+						"black ape",
+						"nigger ape",
+						"chink",
+						"slanteyes",
+						"beaner",
+						"gook",
+						"cracker",
+						"cracKKKa",
+						"whitey",
+						"TwilightSparkle",
```
Comment (PKPenguin321) [inline tf2_bot_detector/cfg/rules.official.json:129]: what?

**Comment by PKPenguin321** [comment]
A repo I work on frequently, https://github.com/tgstation/tgstation, nearly got permabanned by github for just having slurs scattered about in a few places, even when used in a non-racist context. See https://github.com/tgstation/tgstation/commit/8dbf1cf4ec8dd1d90988ce24d3986833197ed097
Before this is pushed, and in fact, ASAP, the maintainers of this repo should contact github to preemptively clarify the use of slurs in the codebase so that people behind the TF2 bots can't report this repo.

Side note, does this program auto-kick people just for being racist? I'd rather preserve my kick cooldown for actual hackers if that's the case, so I'd like this clarified, perhaps in the readme.

**Pull request closed without merging**